### PR TITLE
fix(input): Fix step property rendering errors.

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #2091

## Summary

fix(input): Fix step property rendering errors.

### CODE REVIEW

**Comment by driskull** [comment]
@eriklharper can we make 'step' to be only a number so that those stencil warnings are no longer reported? 

It seems like `step: any` is just another way for step to be `1` when the type is a number. It seems like we can do that internally, right?

Also, what is the reasoning for `maxString` and `minString`? Seems like we could just pass the number to the input instead of maxString/minString.

File: src/components/calcite-input/calcite-input.e2e.ts [hunk 1]
```diff
@@ -311,7 +311,7 @@ describe("calcite-input", () => {
     expect(await element.getProperty("value")).toBe("12");
     expect(await element.getProperty("min")).toBe(10);
     expect(await element.getProperty("max")).toBe(15);
-    expect(await element.getProperty("step")).toBe("1");
```
Comment (driskull) [inline src/components/calcite-input/calcite-input.e2e.ts:314]: This was kind of weird to be a string. If min/max are numbers why can't step be?

**Comment by eriklharper** [comment]
> @eriklharper can we make 'step' to be only a number so that those stencil warnings are no longer reported?
> 
> It seems like `step: any` is just another way for step to be `1` when the type is a number. It seems like we can do that internally, right?
> 
> Also, what is the reasoning for `maxString` and `minString`? Seems like we could just pass the number to the input instead of maxString/minString.

@ethanbdev reported [this issue](https://github.com/Esri/calcite-components/issues/1485) which is why support for `any` is in place.  We could add a custom boolean prop like `stepAny` that could be used to both satisfy the requirements in #1485 and get around this annoying console issue, but ultimately, [this is a Stencil issue](https://github.com/ionic-team/stencil/issues/2814) that they just need to fix already, IMHO.

File: src/components/calcite-input/calcite-input.e2e.ts (hunk 1 above)
Comment (eriklharper) [reply src/components/calcite-input/calcite-input.e2e.ts:314]: Yeah, if its checking the property, it should be a number, not a string.  Only the attribute would be set to a string.

**Comment by eriklharper** [comment]
> Also, what is the reasoning for `maxString` and `minString`? Seems like we could just pass the number to the input instead of maxString/minString.

No idea why `maxString` and `minString` have to exist.  Stencil/JSX should be able to handle passing the number value to the input.

File: src/components/calcite-input/calcite-input.e2e.ts [hunk 2]
```diff
@@ -252,7 +252,7 @@ describe("calcite-input", () => {
   it("correctly increments and decrements value by one when any is set for step", async () => {
     const page = await newE2EPage();
     await page.setContent(`
-    <calcite-input type="number" step="any" value="5.5"></calcite-input>
+    <calcite-input type="number" value="5.5"></calcite-input>
```
Comment (eriklharper) [inline src/components/calcite-input/calcite-input.e2e.ts:255]: If we're really gonna drop support for `any` this test should just get deleted.

File: src/components/calcite-input/calcite-input.tsx [hunk 3]
```diff
@@ -244,6 +244,16 @@ export class CalciteInput {
...
   @State() localizedValue: string;
 
+  @State() computedStep: number;
+
+  @Watch("type")
+  @Watch("step")
+  setComputedStep(): void {
+    const { step, type } = this;
+
+    this.computedStep = step ? Math.abs(step) : type === "number" ? 1 : null;
```
Comment (eriklharper) [inline src/components/calcite-input/calcite-input.tsx:254]: ☝️ `step` supports floats in addition to integers, so we probably shouldn't do `Math.abs` here.  This is particularly relevant for making sure up and down arrows increment properly when decimals are expected in the input and the user wants a finer-grained control over the stepped values.

**Review by eriklharper — CHANGES_REQUESTED** [review]
I think rather than totally ditching `step="any"` support we should provide a prop that allows setting that special value for the use cases where it is needed.  I can envision something like when a `stepAny` prop is set to `true` it will bypass the `this.step` property entirely to achieve its intended behavior, but by default it will stay `false` and we can permanently convert it to be a number all the time.

**Pull request closed without merging**